src/main/java/org/minions/devfund/alejandro/Squarelotron.java

```diff
@@ -0,0 +1,176 @@
+package org.minions.devfund.alejandro;
+
+/**
+ * Squareotron class.
+ */
+public class Squarelotron {
+
+    private int[][] squarelotron;
+    private int size;
+
+    /**
+     * It fills the 2-dimensional array with the numbers 1 to n squared, in order.
+     * It also sets the size instance variable to be n.
+     * @param n The size for the matrix
+     */
+    public Squarelotron(int n) {
+        this.size = n;
+        n = 1;
+        squarelotron = new int[this.size][this.size];
+        for (int x = 0; x < this.size; x++) {
+            for (int y = 0; y < this.size; y++) {
+                squarelotron[x][y] = n++;
+            }
+        }
+    }
+
+    /**
+     * This method performs the Upside-Down Flip of the squarelotron,
+     * and returns the new squarelotron. The original squarelotron should not be modified (we will check for this).
+     * @param ring the ring that will be flipped.
+     * @return A modified instance of a new Squarelotron.
+     */
+    public Squarelotron upsideDownFlip(int ring) {
+
+        Squarelotron resultSquare = new Squarelotron(this.size);
+        int[][] swappedMatrix = this.getSquarelotron();
+
+        if (ring > 0 && ring <= this.size / 2) {
+
+            swappedMatrix = swapMatrix(swappedMatrix);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment; "swappedMatrix" already holds the assigned value along all execution paths. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4165)